### PR TITLE
Add PhpdocSimplifyArrayKeyFixer

### PR DESCRIPTION
**Pull request by spawnia**
## Summary

Adds a custom fixer that simplifies `array<array-key, T>` to `array<T>`.

Since `array-key` is equivalent to `int|string` and arrays naturally have `int|string` keys, specifying this explicitly is redundant. The fixer also handles `array<int|string, T>` and `array<string|int, T>` variants.

## Test plan

- [x] Install dev branch in a project: `composer require mll-lab/php-cs-fixer-config:dev-phpdoc-simplify-array-key-fixer --dev`
- [x] Run php-cs-fixer and verify it transforms `array<array-key, T>` to `array<T>`
- [x] Verify PHPStan still passes after the transformation

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/Fixer/PhpdocSimplifyArrayKeyFixer.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            ]
+        );
+    }
+
+    public function isCandidate(Tokens $tokens): bool
+    {
+        return $tokens->isTokenKindFound(T_DOC_COMMENT);
+    }
+
+    protected function normalize(string $type): string
```
Comment (simbig) [inline src/Fixer/PhpdocSimplifyArrayKeyFixer.php:49]: test?

Comment (spawnia) [reply src/Fixer/PhpdocSimplifyArrayKeyFixer.php:49]: Guter Punkt. Es ist dabei noch ein Edge-Case aufgefallen (multi-line). Ich gebe zu dass ich in diesem Fall den Code nicht mal gelesen hatte 😆 Habe das Projekt nun an php-utils angeglichen.

**Review by simbig — APPROVED** [review]
